vault/resource_consul_secret_backend_role.go

```diff
@@ -38,53 +47,95 @@ func consulSecretBackendRoleResource() *schema.Resource {
 					Type: schema.TypeString,
 				},
 			},
+			"max_ttl": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Description: "Maximum TTL for leases associated with this role, in seconds.",
+				Default:     0,
+			},
+			"ttl": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Description: "Specifies the TTL for this role.",
+				Default:     0,
+			},
+			"token_type": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Specifies the type of token to create when using this role. Valid values are \"client\" or \"management\".",
+				Default:     "client",
+			},
+			"local": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Description: "Indicates that the token should not be replicated globally and instead be local to the current datacenter.",
+				Default:     false,
+			},
 		},
 	}
 }
 
-func consulSecretBackendRoleCreate(d *schema.ResourceData, meta interface{}) error {
+func consulSecretBackendRoleWrite(d *schema.ResourceData, meta interface{}) error {
 	client := meta.(*api.Client)
 
+	backend := d.Get("backend").(string)
 	name := d.Get("name").(string)
-	path := d.Get("path").(string)
-	policies := d.Get("policies").([]interface{})
 
-	reqPath := consulSecretBackendRolePath(path, name)
+	path := consulSecretBackendRolePath(backend, name)
+
+	policies := d.Get("policies").([]interface{})
 
 	payload := map[string]interface{}{
 		"policies": policies,
 	}
 
-	d.Partial(true)
-	log.Printf("[DEBUG] Configuring Consul secrets backend role at %q", reqPath)
+	if v, ok := d.GetOkExists("max_ttl"); ok {
+		payload["max_ttl"] = v
+	}
+	if v, ok := d.GetOkExists("ttl"); ok {
+		payload["ttl"] = v
+	}
+	if v, ok := d.GetOkExists("token_type"); ok {
+		payload["token_type"] = v
+	}
+	if v, ok := d.GetOkExists("local"); ok {
+		payload["local"] = v
+	}
 
-	d.SetId(path + "," + name)
+	log.Printf("[DEBUG] Configuring Consul secrets backend role at %q", path)
 
-	if _, err := client.Logical().Write(reqPath, payload); err != nil {
-		return fmt.Errorf("error writing role configuration for %q: %s", reqPath, err)
+	if _, err := client.Logical().Write(path, payload); err != nil {
+		return fmt.Errorf("error writing role configuration for %q: %s", path, err)
 	}
-	d.SetPartial("name")
-	d.SetPartial("path")
-	d.SetPartial("policies")
-	d.Partial(false)
 
-	return nil
+	d.SetId(path)
```

Comment:
Since this will be a different path now, how will this effect folks migrating from a previous version to this?

Reply:
Yeah I'm guessing this will break? Is there a way to migrate IDs in a provider too?
I kinda went for the nuclear option initially here, but am happy to make changes to improve backwards compatibility.

The current ID is completely different from other role resources, which was a bit confusing when I initially enabled importing and tried to import existing roles.

Reply:
I'm open to updating it, or not updating it. I think you make a good point. If we go that way, it would be best to make the upgrade path for folks seamless, mainly because I should only make breaking changes in major releases of the provider, and we don't have a major release planned soon. 

If we decide to go with changing it, on the question of how, it may be possible using the `StateUpgrader` on the resource. Perhaps it could be used to check for old paths and upgrade them to the new paths. Another idea might be supporting both paths in the Exists and Read funcs, and then upgrading to the new path.

Reply:
I've added code to handle the old format IDs. I'm not sure how to write tests for this though.
I can't see a way to get a state with the old ID format in the tests?

It does work for me in manual testing though